app/controllers/submit.js

```diff
@@ -211,95 +201,101 @@ export default Ember.Controller.extend(Analytics, BasicsValidations, NodeActions
-        // Does the pending title differ from the title already saved?
         return this.get('node.title') !== this.get('nodeTitle');
     }),
-    uploadChanged: Ember.computed('preprintFileChanged', 'titleChanged', function() {
-        // Are there any unsaved changes in the upload section?
-        return this.get('preprintFileChanged') || this.get('titleChanged');
-    }),
+
+    // Are there any unsaved changes in the upload section?
+    uploadChanged: Ember.computed.or('preprintFileChanged', 'titleChanged'),
+
     ////////////////////////////////////////////////////
     // Fields used in the "basics" section of the form.
     ////////////////////////////////////////////////////
+
+    // Pending abstract
     basicsAbstract:  Ember.computed('node.description', function() {
-        // Pending abstract
         let node = this.get('node');
         return node ? node.get('description') : null;
     }),
+
+    // Does the pending abstract differ from the saved abstract in the db?
     abstractChanged: Ember.computed('basicsAbstract', 'node.description', function() {
-        // Does the pending abstract differ from the saved abstract in the db?
         let basicsAbstract = this.get('basicsAbstract');
         return basicsAbstract !== null && basicsAbstract.trim() !== this.get('node.description');
     }),
+
+    // Pending tags
     basicsTags: Ember.computed('node', function() {
-        // Pending tags
-        let node = this.get('node');
-        let newTags = null;
-        if (node != null) {
-            newTags = node.get('tags').slice(0).map(fixSpecialChar);
-        }
-        return node ? newTags : Ember.A();
-    }),
-    tagsChanged: Ember.computed('basicsTags', 'node.tags', function() {
-        // Does the list of pending tags differ from the saved tags in the db?
-        let basicsTags = this.get('basicsTags');
-        let nodeTags = this.get('node.tags');
-        let changed = false;
-        if (basicsTags && nodeTags) {
-            changed = !(basicsTags.length === nodeTags.length && basicsTags.every((v, i)=> fixSpecialChar(v) === fixSpecialChar(nodeTags[i])));
-        }
-        return changed;
+        const node = this.get('node');
+
+        return node ? node.get('tags').map(fixSpecialChar) : Ember.A();
     }),
-    basicsDOI: Ember.computed('model', function() {
-        // Pending DOI
-        return this.get('model.doi');
+
+    // Does the list of pending tags differ from the saved tags in the db?
+    tagsChanged: Ember.computed('basicsTags.@each', 'node.tags', function() {
+        const basicsTags = this.get('basicsTags');
+        const nodeTags = this.get('node.tags');
+
+        return basicsTags && nodeTags &&
+            (
+                basicsTags.length !== nodeTags.length ||
+                basicsTags.some(
+                    (v, i) => fixSpecialChar(v) !== fixSpecialChar(nodeTags[i])
+                )
+            );
     }),
+
+    basicsDOI: Ember.computed.or('model.doi'),
```

Comment:
For some reason, when using `Ember.computed.alias` the validation breaks, but `Ember.computed.or` works just fine--weird